File: JAVA/1704. Determine if String Halves Are Alike.java
class Solution {
    public boolean halvesAreAlike(String s) {
        int c1 = 0;
        int c2 = 0;

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            String charStr = String.valueOf(ch);

            if (("a".equals(charStr) || 
                 "e".equals(charStr) || 
                 "i".equals(charStr) || 
                 "o".equals(charStr) || 
                 "u".equals(charStr) || 
                 "A".equals(charStr) || 
                 "E".equals(charStr) || 
                 "I".equals(charStr) || 
                 "O".equals(charStr) || 
                 "U".equals(charStr)) && (i < s.length() / 2)) {
                c1++;
            } else if ("a".equals(charStr) || 
                       "e".equals(charStr) || 
                       "i".equals(charStr) || 
                       "o".equals(charStr) || 
                       "u".equals(charStr) || 
                       "A".equals(charStr) || 
                       "E".equals(charStr) || 
                       "I".equals(charStr) || 
                       "O".equals(charStr) || 
                       "U".equals(charStr)) {
                c2++;
            }
        }

        return c1 == c2;
    }
}

File: C++/1704. Determine if String Halves Are Alike.cpp
class Solution {
public:
    bool halvesAreAlike(string s) 
    {
        int c1=0,c2=0;
        int j=s.length()/2;
        vector<int>v,a;
        for(int i=0;i<s.length()/2;i++)
        {
           if(s[i]=='a'||s[i]=='e'||s[i]=='i'||s[i]=='o'||s[i]=='u'||s[i]=='A'||s[i]=='E'||s[i]=='I'||s[i]=='O'||s[i]=='U')
             c1++;
        }
         for(int i=j;i<s.length();i++)
        {
           if(s[i]=='a'||s[i]=='e'||s[i]=='i'||s[i]=='o'||s[i]=='u'||s[i]=='A'||s[i]=='E'||s[i]=='I'||s[i]=='O'||s[i]=='U')
             c2++;
        }
        if(c1==c2)
        return true;
        else
        return false;
            
    }
};
